File: rock_paper_scissors.py
import random

cpu = 0
usr = 0

while (1):
    # prints title
    print("Rock Paper Scissors")
    print("cpu " + str(cpu) + " | usr " + str(usr))
    
    # User Plays
    userin = ""
    userchoice = 0
    print("User chooses ", end="")
    input(userin)
    userin = userin.lower()
    if (userin == "rock"):
        userchoice = 0
    elif (userin == "paper"):
        userchoice = 1
    elif (userin == "scissor"):
        userchoice = 2
    
    # Computer Plays
    cpurand = random.randint(0, 2)
    if (cpurand == 0):
        print('Computer choose rock')
    elif (cpurand == 1):
        print('Computer choose paper')
    elif (cpurand == 2):
        print ('Computer choose scissors')
        
    # Scoring
    if (userchoice == cpurand):
        print('Draw')
    if ((userchoice == 0 and cpurand == 2) or (userchoice == 1 and cpurand == 0) or (userchoice == 2 and cpurand == 1)):
        print('User Wins!')
        usr += 1
    if ((cpurand == 0 and userchoice == 2) or (cpurand == 1 and userchoice == 0) or (cpurand == 2 and userchoice == 1)):
        print('Cpu Wins! :(')
        cpu += 1

File: rock_paper_scissors.c
#include <stdlib.h>
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

int main()
{
    int cpu = 0;
    int usr = 0;
    
    while (true) {
        // prints title
        printf("-------------------\n");
        printf("Rock Paper Scissors\n");
        printf("cpu %d | usr %d\n", cpu, usr);
        
        // user plays
        char userin[10];
        int userchoice = 0;
        printf("User choses ");
        scanf("%s", userin);

        if (!strcmp(userin, "rock") || !strcmp(userin, "ROCK") || !strcmp(userin, "Rock")){
            userchoice = 0;
        }
        else if (!strcmp(userin, "paper") || !strcmp(userin, "PAPER") || !strcmp(userin, "Paper")){
            userchoice = 1;
        }
        else if (!strcmp(userin, "scissor") || !strcmp(userin, "SCISSOR") || !strcmp(userin, "Scissor")){
            userchoice = 2;
        }
        else {
            printf("ERROR INVALID INPUT\n");
            continue;
        }
        
        // cpu plays
        int cpuchoice = rand() % 3;
        printf("Computer choses ");
        if (cpuchoice == 0){
            printf("rock\n");
        }
        else if (cpuchoice == 1){
            printf("paper\n");
        }
        else if (cpuchoice == 2){
            printf("scissor\n");
        }
        
        // Scoring
        if (userchoice == cpuchoice){
            printf("DRAW\n");
        }
        else if (
            (userchoice == 0 && cpuchoice == 1) || // rock vs paper
            (userchoice == 1 && cpuchoice == 2) || // paper vs scissor
            (userchoice == 2 && cpuchoice == 0)    // scissor vs rock
            ){
                printf("CPU wins!\n");
                cpu += 1;
        }
        else if (
            (userchoice == 0 && cpuchoice == 2) || // rock vs scissor
            (userchoice == 1 && cpuchoice == 0) || // paper vs rock
            (userchoice == 2 && cpuchoice == 1)    // scissor vs paper
            ){
                printf("USER wins!\n");
                usr += 1;
        }
    }
}
